android/TodokTodok/app/src/main/java/com/team/todoktodok/presentation/view/discussions/vm/DiscussionsViewModel.kt

```
@@ -75,21 +78,23 @@
     private fun onUiEvent(
         discussions: List<Discussion>,
         filter: DiscussionFilter,
     ) {
-        _uiEvent.value =
+        val event =
             when {
                 discussions.isEmpty() && filter == DiscussionFilter.ALL -> DiscussionsUiEvent.ShowNotHasAllDiscussions
                 discussions.isEmpty() && filter == DiscussionFilter.MINE -> DiscussionsUiEvent.ShowNotHasMyDiscussions
                 filter == DiscussionFilter.ALL -> DiscussionsUiEvent.ShowHasAllDiscussions
                 filter == DiscussionFilter.MINE -> DiscussionsUiEvent.ShowHasMyDiscussions
                 else -> throw IllegalArgumentException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)